### ISSUE TITLE
test networks have more branches than training networks?

### ISSUE THREAD

**Issue by Cogito2012** [zhaoweicai/cascade-rcnn#31]
I have visualized your provided prototxt files for both training and test networks. I found that the test network has additional fully-connected layers which does not exist in training network, i.e.:

- stage2:  [roi_pool_2nd] --> FC6/FC7 --> [cls_prob_1st_2nd] -->...

- stage3: [roi_pool_3rd] --> FC6/FC7/FC/Softmax --> [cls_prob_1st_3rd] -->...
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[roi_pool_3rd] --> FC6/FC7/FC/Softmax --> [cls_prob_2nd_3rd] -->...

I was confused that if these additional branches are not trained, how can they be used in test phase? Or the parameters of these branches are simply copied from cls prediction branch in RCNN? In this case, will these branches in the same stage predict the same cls_prob?

**Comment by zhaoweicai** [comment]
The new layers added at inference time share parameters of the trained heads. This is for ensemble of three detection heads. You can have a look at the description of Table 2 in the paper.
